packages/contracts-bedrock/scripts/Deploy.s.sol

```diff
@@ -487,30 +512,32 @@ contract Deploy is Deployer {
 
     /// @notice Deploy the SystemConfig
     function deploySystemConfig() public broadcast returns (address addr_) {
-        SystemConfig config = new SystemConfig{ salt: implSalt() }();
+        ResourceMetering.ResourceConfig memory defaultConfig = Constants.DEFAULT_RESOURCE_CONFIG();
+        uint64 minimumGasLimit = uint64(defaultConfig.maxResourceLimit) + uint64(defaultConfig.systemTxMaxGas);
+        SystemConfig config = new SystemConfig({
+            _owner: address(0xdEaD),
+            _overhead: 0,
+            _scalar: 0,
+            _batcherHash: bytes32(0),
+            _gasLimit: minimumGasLimit,
+            _unsafeBlockSigner: address(0),
+            _config: defaultConfig
+        });
 
         require(config.owner() == address(0xdEaD));
         require(config.overhead() == 0);
         require(config.scalar() == 0);
         require(config.unsafeBlockSigner() == address(0));
         require(config.batcherHash() == bytes32(0));
-        require(config.gasLimit() == 1);
+        require(config.gasLimit() == minimumGasLimit);
 
         ResourceMetering.ResourceConfig memory resourceConfig = config.resourceConfig();
-        require(resourceConfig.maxResourceLimit == 1);
-        require(resourceConfig.elasticityMultiplier == 1);
-        require(resourceConfig.baseFeeMaxChangeDenominator == 2);
-        require(resourceConfig.systemTxMaxGas == 0);
-        require(resourceConfig.minimumBaseFee == 0);
-        require(resourceConfig.maximumBaseFee == 0);
-
-        require(config.l1ERC721Bridge() == address(0));
-        require(config.l1StandardBridge() == address(0));
-        require(config.l2OutputOracle() == address(0));
-        require(config.optimismPortal() == address(0));
-        require(config.l1CrossDomainMessenger() == address(0));
-        require(config.optimismMintableERC20Factory() == address(0));
-        require(config.startBlock() == type(uint256).max);
+        require(resourceConfig.maxResourceLimit == defaultConfig.maxResourceLimit);
+        require(resourceConfig.elasticityMultiplier == defaultConfig.elasticityMultiplier);
+        require(resourceConfig.baseFeeMaxChangeDenominator == defaultConfig.baseFeeMaxChangeDenominator);
```

Comment:
I thought we were using the modularized library for all of the checks?

Reply:
This is what's on `develop`. It looks like the PR only used the `ChainAssertions` lib in the `initializeX()` functions.

Reply:
After looking into this more, I think it would be best done in a separate PR.

Some of the checks are straightforward, but for other contracts it's fairly involved. 

1. we'll need a new `_implementations()` getter that serves the same purpose as `_proxies()`
2. for contracts where the proxy and implementation have a different set of checks, we'll need two new functions,
  i.e `checkSystemConfig()`, which be split into `checkSystemConfigImpl`, `checkSystemConfigProxy` and `checkSystemConfigCommon`.

Or there might be better ways to do it, but either way it's going to open up new territory for bike shedding.

Reply:
Sure thing, [219](https://github.com/ethereum-optimism/client-pod/issues/219).